libcfnet/tls_client.c

```diff
@@ -340,7 +340,8 @@ int TLSTry(ConnectionInfo *conn_info)
             sleep(1);
             remaining_tries--;
         }
-    }
+    } while (!connected && should_retry);
```

Comment:
I don't like `do-while` and LGTM should now be OK with this code (there used to be a false-positive warning).